Apollo/Components/GraphDial.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+
+using Avalonia;
+using Avalonia.Controls;
+using Avalonia.Controls.Shapes;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using Avalonia.Markup.Xaml;
+using Avalonia.Media;
+using Avalonia.Threading;
+
+using Apollo.Core;
+using Apollo.Structures;
+using Avalonia.Media.Imaging;
+using Avalonia.Platform;
+
+namespace Apollo.Components {
+    public class GraphDial: Dial {
+        
+        public delegate void DialBilateralChangedEventHandler(GraphDial sender, bool NewValue, bool? OldValue);
+        public event DialBilateralChangedEventHandler BilateralChanged;
+        
+        bool _isBilateral;
+        public bool IsBilateral{
+            get => _isBilateral;
+            set {
+                _isBilateral = value;
+                DrawArcAuto();
+                BilateralChanged?.Invoke(this, IsBilateral, !IsBilateral);
+            }
+        }
+        
+        protected override void InitializeComponent() {
+            AvaloniaXamlLoader.Load(this);
+            
+            ArcCanvas = this.Get<Canvas>("ArcCanvas");
+            
+            ArcBase = this.Get<Path>("ArcBase");
+            Arc = this.Get<Path>("Arc");
+
+            Display = this.Get<TextBlock>("Display");
+            TitleText = this.Get<TextBlock>("Title");
+
+            Input = this.Get<TextBox>("Input");
+        }
+        
+        public GraphDial(){
+            InitializeComponent();
+            
+            Minimum = -2;
+            Maximum = 2;
+            Round = 1;
+            
+            IsBilateral = false;
+            
+            Arc.StrokeThickness = 2 * Scale;
+            
+            ArcBase.StrokeThickness = 4 * Scale;
+            ArcBase.Stroke = (IBrush)Application.Current.Styles.FindResource("ThemeForegroundLowBrush");
+            
+            ModeChanged += (bool _, bool? __) => IsBilateral = !IsBilateral;
+        }
+        
+        protected void Graph_Unloaded(object sender, VisualTreeAttachmentEventArgs e){
+            Unloaded(sender, e);
+            
+            BilateralChanged = null;
+        }
+        
+        public override void DrawArcAuto() {
+            Display.Text = (Enabled || !DisplayDisabledText)? ValueString : DisabledText;
+            
+            if(!Enabled){
+                Arc.Stroke = (IBrush)Application.Current.Styles.FindResource("ThemeForegroundLowBrush");
+                ArcBase.IsVisible = false;
+                return;
+            }
+            
+            ArcBase.IsVisible = true;
+            
+            
+            if (IsBilateral) DrawArcBilateral();
+            else DrawArcQuad();
+        }
+        
+        protected override void DrawArc(Path Arc, double value, bool overrideBase, string color = "ThemeAccentBrush"){
```

Comment:
yeah this is p much entirely flipped on its head, ill have to refactor this

Reply:
aight